javamelody-core/src/main/java/net/bull/javamelody/internal/model/Collector.java

```diff
@@ -533,6 +546,35 @@ private void collectOtherJavaInformations(List<JavaInformations> javaInformation
 		// différenciation des indisponibilités prévues de celles non prévues
 	}
 
+	private void collectCurrentRequests(List<JavaInformations> javaInformationsList) {
+		if (currRequestMinDurationLog == null) return;
+		if (metricsPublishers == null) return;
+
+		List<ThreadInformations> threadInformationsList = javaInformationsList.get(0).getThreadInformationsList();
+		try {
+			List<CounterRequestContext> rootCurrentContextList = getRootCurrentContexts(counters);
+			for (final CounterRequestContext rootCurrentContext : rootCurrentContextList) {
+				if (rootCurrentContext.getDuration(System.currentTimeMillis()) < (Integer.parseInt(currRequestMinDurationLog) * 1000)) return;
+				List fullStackTrace = threadInformationsList.stream()
+						.filter(threadInformations -> threadInformations.getId() == rootCurrentContext.getThreadId())
+						.collect(Collectors.toList())
+						.get(0).getStackTrace();

```

Comment:
javamelody should be compatible with java 7. It will happen sometimes that current requests are not synchronized with current threads, so IndexOutOfBoundsException here.